src/basic/macro.h

```diff
@@ -634,4 +634,6 @@ static inline int __coverity_check_and_return__(int condition) {
                 _copy;                                                  \
         })
 
+#define SIZE_ADD(x, y) ((x) >= SIZE_MAX - (y) ? SIZE_MAX : (x) + (y))
```

Comment:
double eval sucks though, no? we are so far pretty good at avoiding double eval in the macros in macro.h, and i think we should maintain that standard for macros in macro.h

I mean, your use of this macro already nests two invocations inside each other, the double eval already starts being ugly there, but hopefully the compiler fixes that...

use a static inline func or  `({ … })`? I'd probably use static inline func, it's simple given that you know the types and don#t ned to be generic, and thus need no UNIQ_T shit.

(yeah, that means this no longer a constant expression, but I value double-eval-freeness more than constant expressionness, I must say)

Reply:
Updated with an inline function:
```diff
-#define SIZE_ADD(x, y) ((x) >= SIZE_MAX - (y) ? SIZE_MAX : (x) + (y))
+static inline size_t size_add(size_t x, size_t y) {
+        return y >= SIZE_MAX - x ? SIZE_MAX : x + y;
+}
```